src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,175 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+
+    private class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (first == null) {
+            first = new Node<>(null, value, null);
+            return true;
+        }
+        Node<T> element = first;
+        while (element.next != null) {
+            element = element.next;
+        }
+        Node<T> newNode = new Node<T>(element, value, null);
+        element.next = newNode;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (first == null) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNodeByIndex(index);
 
+        if (currentNode == null) {
+            Node<T> prevCurrent = getNodeByIndex(index - 1);
```

Comment:
Why are you looking for a previous node? You can contact her through ---  currentNode.prev